src/shared/dlt_common.c

```diff
@@ -1825,7 +1825,7 @@ void dlt_log_init(int mode)
 
     if (logging_mode == DLT_LOG_TO_FILE) {
         /* internal logging to file */
-        logging_handle = fopen(logging_filename, "a");
+        logging_handle = fopen(logging_filename, file_mode);
```

Comment:
I think we could create an internal function, something like const char *dlt_set_log_to_file_mode(int overwrite_flag) instead of trying to add 1 more argument.
How do you think?